### PR TITLE
Welcome page

### PR DESCRIPTION
**Pull request by e1ad**
- Add a new Welcome route
- Option to add more players
- Select a opener category

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
     "@types/react-dom": "^18.3.0",
     "@vitejs/plugin-react-swc": "^3.5.0",
     "autoprefixer": "^10.4.20",
+    "daisyui": "^4.12.13",
```
Comment (e1ad) [inline package.json:27]: Lets take about it

File: src/views/Welcome/Welcome.tsx [hunk 2]
```diff
@@ -1,5 +1,64 @@
-import React from 'react';
+import React, {useCallback} from 'react';
+import {useNavigate} from 'react-router-dom';
+import {useGame} from '@contexts/game.context';
+import {MAX_PLAYERS, PlayerColorBank} from '@components/app/consts';
+import openings from '@assets/openings.json';
+
+const cetegories = Object.keys(openings);
```
Comment (codelicous) [inline src/views/Welcome/Welcome.tsx:7]: categories - typo

File: src/views/Welcome/Welcome.tsx [hunk 3]
```diff
@@ -1,5 +1,64 @@
...
+            navigate('/');
+        }
+    }, [players, navigate]);
+
+    const addPlayer = useCallback(() => {
+        const value = inputRef.current?.value;
+        if (value) {
+            setConfig((prevState) => ({
+                    ...prevState,
+                    players: [...prevState.players, { name: value, id: prevState.players.length + 1, color: PlayerColorBank.player3}]
```
Comment (codelicous) [inline src/views/Welcome/Welcome.tsx:30]: Several things 
1. should have mentioned it before , i think it makes sense here to have a max limit of 5 players to the game and enforce it at this stage, and a minimum of 2. 
2. all added players get the same player color it should be added dynamically.
3. For some reason from what i saw the third added players does not have a progress colored bar running.
4. I think we can remove the hard coded players and for development and debugging purposes (or perhaps also a good UI decision) have a quick play button - that can be extracted to a different ticket.

File: src/views/Welcome/Welcome.tsx [hunk 4]
```diff
@@ -1,5 +1,64 @@
...
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    const onCategoryChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setConfig((prevState) => ({...prevState, openerCategory: event.target.value as Game['openerCategory']}));
+    }, [setConfig]);
+
+    const onPlayClick = useCallback(() => {
+        const isValid = players.every((player) => player.name.trim().length);
+        if (isValid) {
+            navigate('/');
```
Comment (codelicous) [inline src/views/Welcome/Welcome.tsx:21]: I think the welcome page should be rerouted to from empty and the game should be assigned another route , what do you think ?

**Review by codelicous — COMMENTED** [review]
Nice work , as you mentioned i think we should talk about the library and further stpes forward

**Comment by ofergroman** [comment]
@e1ad There's an issue on initial game start - for some reason the first player has two consecutive turns

File: src/views/Welcome/Welcome.tsx (hunk 3 above)
Comment (ofergroman) [reply src/views/Welcome/Welcome.tsx:30]: Maybe we can add a param in the url - ?debug=true

Comment (codelicous) [reply src/views/Welcome/Welcome.tsx]: maybe,  or "quickplay" with player names player 1 player 2

**Pull request merged**